juju/client/facade.py

```diff
@@ -926,11 +927,11 @@ def generate_definitions(schemas):
     return definitions
 
 
-def generate_facades(schemas):
+def generate_facades(schemas: Dict[str, List[Schema]]) -> Dict[str, Dict[int, codegen.Capture]]:
     captures = defaultdict(codegen.Capture)
 
     # Build the Facade classes
-    for juju_version in sorted(schemas.keys()):
+    for juju_version in sorted(schemas.keys(), key=packaging.version.parse):
```

Comment:
This is nice. It'll break on the "latest" string, right? If so, let's just remove the special casing in `load_schemas` (line 955). Existing code pasted below for convenient reference.

```python
def load_schemas(options):
    schemas = {}
    for p in sorted(glob(options.schema)):
        if 'latest' in p:
            juju_version = 'latest'
        else:
            try:
                juju_version = re.search(JUJU_VERSION, p).group()
            except AttributeError:
                print("Cannot extract a juju version from {}".format(p))
                print("Schemas must include a juju version in the filename")
                raise SystemExit(1)
```

Reply:
Done